File: concurrency/java/Blocking.java
class Blocking {
    public void start() {
        long startTime = System.currentTimeMillis();
        while (true) {
            System.out.println("Rendring UI...." + " On: " + Thread.currentThread().getName());
            long now = System.currentTimeMillis();

            if (now - startTime >= 5000 && now - startTime <= 6000) {
                doRequest();
            }

            sleep(1000);
        }

    }

    private void doRequest() {
        System.out.println("Initiated the request...." + " On: " + Thread.currentThread().getName());
        sleep(4000);
        System.out.println("Finished the request On: " + Thread.currentThread().getName());

    }

    private void sleep(long time) {
        try {
            Thread.sleep(time);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: concurrency/python/blocking.py
import time
from datetime import datetime
import threading


class Blocking:
    def start(self):
        startTime = datetime.now()

        while True:
            print(f"Rendring UI on... ${threading.current_thread().name}")
            now = datetime.now()
            elapsed_time = (now - startTime).total_seconds()

            if elapsed_time >= 5 and elapsed_time <= 6:
                self.doRequest()

            time.sleep(1)

    def doRequest(self):
        print(f"Initiated the request.... on  {threading.current_thread().name}")
        time.sleep(3)
        print(f"Finished the request.... on {threading.current_thread().name}")
